src/uct/ib/mlx5/gga/gga_mlx5.c

```diff
@@ -625,6 +625,22 @@ uct_gga_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
            uct_ib_iface_is_reachable_v2(tl_iface, params);
 }
 
+static int
+uct_gga_mlx5_ep_is_connected(uct_ep_h tl_ep,
+                             const uct_ep_is_connected_params_t *params)
+{
+    uint32_t addr_qp = 0;
+    uct_gga_mlx5_ep_address_t *ep_addr;
+
+    if (params->field_mask & UCT_EP_IS_CONNECTED_FIELD_EP_ADDR) {
+        ep_addr = (uct_gga_mlx5_ep_address_t*)params->ep_addr;
+        addr_qp = uct_ib_unpack_uint24(ep_addr->qp_num);
+    }
```

Comment:
there is some code duplication with uct_rc_mlx5_base_ep_is_connected
to eliminate it we can create new parent struct for uct_gga_mlx5_ep_address_t and uct_rc_mlx5_ep_address_t - that will contain only qp_num. This struct can be called uct_rc_mlx5_base_ep_address_t.
uct_rc_mlx5_base_ep_is_connected will cast params->ep_addr to uct_rc_mlx5_base_ep_address_t*, if the flag UCT_EP_IS_CONNECTED_FIELD_EP_ADDR is specified.